src/lib.rs

```diff
@@ -238,11 +280,11 @@ pub trait OpenDrainOutput {
 impl<E: Debug, P: OutputPin<Error = E> + InputPin<Error = E>> OpenDrainOutput for P {
     type Error = E;
 
-    fn is_high(&self) -> Result<bool, Self::Error> {
+    fn is_high(&mut self) -> Result<bool, Self::Error> {
```

Comment:
Is `&mut` really needed here?

Reply:
Same question

Reply:
Yes it indeed needs `&mut` https://github.com/rust-embedded/embedded-hal/blob/master/embedded-hal/src/digital.rs#L208